evcxr/src/module.rs

```diff
@@ -265,11 +267,47 @@ overflow-checks = true
         )
     }
 }
+fn tee(mut stream: impl Read) -> Result<String, Error> {
+    let mut out: String = String::new();
+    let mut buff: [u8; 1024] = [0; 1024];
+    // I dont like infinite loops
+    loop {
+        let bytes_read = stream.read(&mut buff)?;
+        if bytes_read == 0 {
+            return Ok(out);
+        }
+        let partial_out = std::str::from_utf8(&buff)?;
```

Comment:
The partial input may be cut of in the middle of an UTF-8 character and thus fail to verify. You could try pushing everything as raw bytes to the buffer and then to have a live view scan from the last byte that was printed until the last newline (which is a single byte big and due to the way UTF-8 works doesn't occur in the middle of any other encoded character) and then use str::from_utf8 on this slice.

Reply:
Not very sure about how likely it is for this to happen. 
But it is a good catch. Thank you.